### PR TITLE
Rafadana/join cancel event

### PR DESCRIPTION
**Pull request by RafadanaM**
### Background: 
- Adding  join, cancel, and also deleting event

### Additional Packages/Dependencies:
- none

### Changes:
- lib/auth/data/auth_repository.dart
- lib/common/components/buttons/custom_text_button.dart
- lib/common/components/confirmation_modal_bottom.dart
- lib/common/config/enum.dart
- lib/common/utils/error_handler.dart
- lib/common/utils/navigator_util.dart
- lib/event/bloc/event_detail_cubit.dart
- lib/event/bloc/event_detail_state.dart
- lib/event/bloc/update_event_detail_cubit.dart
- lib/event/bloc/update_event_detail_state.dart
- lib/event/components/edit_bottom_modal.dart
- lib/event/components/leave_bottom_modal.dart
- lib/event/data/event_detail_data_provider.dart
- lib/event/data/event_detail_model.dart
- lib/event/data/event_detail_repository.dart
- lib/event/data/event_detail_response_model.dart
- lib/event/data/event_participant_model.dart
- lib/get_it.dart
- lib/page/event_detail.dart
- lib/preference/data/preference_model.dart

### How to Implement:
- none

### CODE REVIEW

File: lib/event/bloc/event_detail_cubit.dart [hunk 1]
```diff
@@ -1,16 +1,42 @@
...
   }
+
+  @override
+  Future<void> close() {
+    _streamSubscription.cancel();
+    _eventDetailRepository.dispose();
+    return super.close();
+  }
+
+  deleteEvent(int eventID) {}
```
Comment (YafiAhsanH) [inline lib/event/bloc/event_detail_cubit.dart]: is the deleteEvent supposed to be empty?

File: lib/event/bloc/event_detail_state.dart [hunk 2]
```diff
@@ -1,24 +1,44 @@
...
+
+// class EventDetailSuccessState extends EventDetailState {
+//   final EventDetailResponseModel eventDetailResponseModel;
+//   const EventDetailSuccessState({required this.eventDetailResponseModel});
+// }
+
+// class EventDetailErrorState extends EventDetailState {
+//   final BaseException error;
+//   const EventDetailErrorState({required this.error});
+// }
```
Comment (YafiAhsanH) [inline lib/event/bloc/event_detail_state.dart]: A lot of comments in this file

**Review by YafiAhsanH — CHANGES_REQUESTED** [review]

**Review by oliver-mf — APPROVED** [review]
looks good

**Review by YafiAhsanH — APPROVED** [review]
looks good!

**Pull request merged**